spec/views/accounts/_work_details.html.erb_spec.rb

```diff
@@ -0,0 +1,550 @@
+        expect(subject).to have_css(".govuk-summary-list__value")
+      end
+
+      it "shows headteacher status" do
+        expect(subject).to have_css(".govuk-summary-list__key", text: "Headteacher")
+        expect(subject).to have_css(".govuk-summary-list__value")
+      end
+
+      it "shows first 5 years of headship when headteacher is yes" do
+        expect(subject).to have_css(".govuk-summary-list__key", text: "First 5 years of headship")
+        expect(subject).to have_css(".govuk-summary-list__value")
+      end
+    end
+
+    context "when course is EHCO but raw_application_data is nil" do
+      let(:ehco_course) { create(:course, :early_headship_coaching_offer) }
+
+      before do
+        application.update!(
+          course: ehco_course,
+          raw_application_data: nil,
+        )
+      end
+
+      it "raises an error due to nil raw_application_data (bug in view)" do
+        expect { render partial: "accounts/work_details", locals: { application: application } }.to raise_error(ActionView::Template::Error, /undefined method '\[\]' for nil/)
+      end
+    end
+
+    context "when course is EHCO and ehco_headteacher is not yes" do
+      let(:ehco_course) { create(:course, :early_headship_coaching_offer) }
+
+      before do
+        application.update!(
+          course: ehco_course,
+          raw_application_data: {
+            "npqh_status" => "completed_npqh_not_in_role",
+            "ehco_headteacher" => "no",
+          },
+        )
+        render partial: "accounts/work_details", locals: { application: application }
+      end
+
+      it "does not show first 5 years of headship" do
+        expect(subject).not_to have_css(".govuk-summary-list__key", text: "First 5 years of headship")
+      end
+    end
+  end
+
+  describe "NPQLPM course specific fields" do
+    context "when course is NPQLPM" do
+      let(:npqlpm_course) { create(:course, :leading_primary_mathematics) }
+
+      before do
+        application.update!(
+          course: npqlpm_course,
+          raw_application_data: {
+            "maths_eligibility_teaching_for_mastery" => "yes",
+          },
+        )
+        render partial: "accounts/work_details", locals: { application: application }
+      end
+
+      it "shows maths teaching for mastery question" do
+        expect(subject).to have_css(".govuk-summary-list__key", text: "Completed one year of the primary maths Teaching for Mastery programme?")
+        expect(subject).to have_css(".govuk-summary-list__value")
+      end
+    end
+
+    context "when course is NPQLPM with 'no' answer" do
+      let(:npqlpm_course) { create(:course, :leading_primary_mathematics) }
+
+      before do
+        application.update!(
+          course: npqlpm_course,
+          raw_application_data: {
+            "maths_eligibility_teaching_for_mastery" => "no",
+            "maths_understanding_of_approach" => "some_understanding",
```

Comment:
```suggestion
            "maths_understanding_of_approach" => "taken_a_similar_course",
```
this is not a valid maths_understanding_of_approach_option